sklearn/utils/estimator_checks.py

```diff
@@ -1360,6 +1390,62 @@ def check_classifiers_train(name, classifier_orig):
                 assert_array_equal(np.argsort(y_log_prob), np.argsort(y_prob))
 
 
+def check_outliers_train(name, estimator_orig):
+    X, _ = make_blobs(n_samples=300, random_state=0)
+    X = shuffle(X, random_state=7)
+    n_samples, n_features = X.shape
+    estimator = clone(estimator_orig)
+    set_random_state(estimator)
+
+    # fit
+    estimator.fit(X)
+    # with lists
+    estimator.fit(X.tolist())
+
+    y_pred = estimator.predict(X)
+    assert_equal(y_pred.shape, (n_samples,))
+    # training set performance
+    # TODO
+
+    # raises error on malformed input for predict
+    assert_raises(ValueError, estimator.predict, X.T)
+
+    # decision_function agrees with predict
+    decision = estimator.decision_function(X)
+    assert_equal(decision.shape, (n_samples,))
+    dec_pred = (decision >= 0).astype(np.int)
+    dec_pred[dec_pred == 0] = -1
+    assert_array_equal(dec_pred, y_pred)
+
+    # raises error on malformed input for decision_function
+    assert_raises(ValueError, estimator.decision_function, X.T)
+
+    # decision_function is a translation of score_samples
+    y_scores = estimator.score_samples(X)
+    assert_equal(y_scores.shape, (n_samples,))
+    y_dec = y_scores - estimator.offset_
+    assert_array_equal(y_dec, decision)
+    assert_array_equal(np.argsort(y_scores), np.argsort(decision))
+
+    # raises error on malformed input for score_samples
+    assert_raises(ValueError, estimator.score_samples, X.T)
+
+    # contamination parameter (not for OneClassSVM which has the nu parameter)
+    if hasattr(estimator, "contamination"):
+        # proportion of outliers equal to contamination parameter when not
+        # set to 'auto'
+        contamination = 0.1
+        estimator.set_params(contamination=contamination)
+        estimator.fit(X)
+        y_pred = estimator.predict(X)
+        assert_equal(np.mean(y_pred != 1), contamination)
```

Comment:
this should be some kind of approximate equality, no?

Reply:
And surely it's only necessarily true of `fit_predict(X)` if, in some case, the requirement that `fit_predict(X) == fit(X).predict(X)` needs to be broken

Reply:
I forgot this one. You're right this should be an approximate equality. Will do it soon.